### ISSUE TITLE
[Question] Interesting fork of Ansible Playbook by remil1000

### ISSUE THREAD

**Issue by peterzhuamazon** [opensearch-project/ansible-playbook#37]
Hi all,

@saravanan30erd I have noticed @remil1000 has a interesting fork which change a lot of the structure based on our initial setup of ansible playbook, even the support of Debian 11 is added.

@remil1000 we would like to have your changes port back in the upstream we have here.
Is it possible for you to contribute directly to this repo?

Thanks.

cc: @stockholmux

**Comment by bbarani** [comment]
@remil1000, we would like to invite you to join us and contribute to OpenSearch Ansible repository in order to create a robust Ansible playbook for the OpenSearch community. Please feel free to reach out to us for any additional information.

**Comment by saravanan30erd** [comment]
@peterzhuamazon yes, correct and it has interesting changes. But it forked from our old state when we supported only single node installation, and he added multi-node (cluster) installation setup and other changes. Since we added multi-node feature and supporting many OS distros now, it might create lot of conflicts if we try to merge those changes directly. Instead @remil1000 can add missing features and changes one by one here.

**Comment by remil1000** [comment]
Hi,
you're right about the fork & feature addition
I started working on this back in November 2021 and some unexpected delays lead to this very large drift and now the issue of a near impossible merge between your work and mine

a few items I could try to backport are the following:
- PKI management using Ansible modules and no external tools (I have a somehow hard requirement to set this up in airgapped environments)
- bcrypt from plain text password with static salt (idempotency) using Ansible module
- some tweaks for cluster initialization (`initial_master_nodes`)
- idempotency of the whole role

let me know how you would like to proceed, if you have any priorities or current pain points to work on

**Comment by saravanan30erd** [comment]
@remil1000 thanks for the response. we don't have any specific priorities or pain points so you can work on all those above 4 points. Suggestion from my side is, create separate PR for every feature and please make sure the newly added changes supports all our current supported distros (Centos7, RHEL7, Amazon Linux2, Ubuntu 20.04) with minimal complexity.

**Comment by peterzhuamazon** [comment]
I second on @saravanan30erd responses and fully welcome your changes @remil1000. 
Please let us know if you have any questions or concerns that we can help with.
Thanks.

**Comment by saravanan30erd** [comment]
@remil1000 I created the role initially focused only for first time installation, didn't have a time to support upgrade and cluster extend features. so it works perfectly only for cluster creation and doesn't support upgrade (re-run) and cluster extend. To support these, we need to work on the points you mentioned above(especially certs). 

Regarding certs, I am currently using searchguard tool to create certs and configuration templates which is later added to main configuration. I see you have already plan to use Ansible modules for PKI management so we can replace the current mechanism with that and try to solve the upgrade or re-run issues. Otherwise we can use the same external tool, and modify to support re-run and node addition as this tool supports to not overwrite if the certs already present in outputs folder and I never tried it since still didn't start to work for upgrade or re-run support.

Regarding variable prefix, sure we can do that.
